storage/range.go

```diff
@@ -235,20 +238,17 @@ func (r *Range) String() string {
 	return fmt.Sprintf("range=%d (%s-%s)", r.Desc().RaftID, r.Desc().StartKey, r.Desc().EndKey)
 }
 
-// start begins gossiping loop in the event this is the first
-// range in the map and gossips config information if the range
-// contains any of the configuration maps.
-func (r *Range) start(stopper *util.Stopper) {
-	r.stopper = stopper
-	// TODO(spencer): gossiping should only commence when the range gains
-	// the leader lease and it should stop when the range no longer holds
-	// the leader lease.
-	r.maybeGossipClusterID()
-	r.maybeGossipFirstRange()
-	r.maybeGossipConfigs(configDescriptors...)
-	// Only start gossiping if this range is the first range.
+// start begins by gossiping the cluster ID (only happens if the range
+// is the first in the key space). This is necessary to get the cluster
+// started from scratch.
+func (r *Range) start() {
 	if r.IsFirstRange() {
-		r.startGossip()
+		r.startGossip(r.rm.Stopper())
+	}
+	if r.Desc().StartKey.Less(engine.KeySystemMax) {
```

Comment:
is that the canonical check? Would've expected something like `StartKey.Less(EndKey)`.